src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -109,24 +108,28 @@ String metricName(MetricName metricName) {
         return prefix + '_' + group + '_' + name;
     }
 
-    static MetricFamilySamples convert(String name, String help, KafkaMetric metric, Map<String, String> labels) {
-        Object value = metric.metricValue();
-        if (!(value instanceof Number)) {
-            // Prometheus only accepts numeric metrics.
-            // Kafka gauges can have arbitrary types, so skip them for now
-            // TODO move non-numeric values to labels
-            return null;
-        }
-        Map<String, String> sanitizedLabels = labels.entrySet().stream()
+    private static MetricFamilySamples convert(String prometheusMetricName, KafkaMetric metric, MetricName metricName) {
+        Map<String, String> sanitizedLabels = metricName.tags().entrySet().stream()
                 .collect(Collectors.toMap(
                         e -> Collector.sanitizeMetricName(e.getKey()),
                         Map.Entry::getValue,
                         (v1, v2) -> {
-                            throw new IllegalStateException("Unexpected duplicate key " + v1);
+                            LOG.warn("Ignoring metric value duplicate key {}", v1);
```

Comment:
Should we have a test to exercise this? Running code coverage it looks like we don't test this. If you want we can tackle that in a separate PR, in that case, please open an issue to track it.

Reply:
@mimaison Are we testing whether the warning is logged or are we testing if there are duplicate keys? A bit unsure. Thanks

Reply:
I'm not sure we necessarily need to check the log line is emitted, but we can check the behavior, i.e. what happens if we add a metric that would cause duplicated labels?

Reply:
New issue created for this: https://github.com/strimzi/metrics-reporter/issues/22